pallets/passkey/src/lib.rs

```diff
@@ -131,28 +146,40 @@ pub mod module {
 	#[pallet::validate_unsigned]
 	impl<T: Config> ValidateUnsigned for Pallet<T>
 	where
-		<T as frame_system::Config>::RuntimeCall: Dispatchable<Info = DispatchInfo>,
+		<T as frame_system::Config>::RuntimeCall:
+			From<Call<T>> + Dispatchable<Info = DispatchInfo, PostInfo = PostDispatchInfo>,
 	{
 		type Call = Call<T>;
 		fn validate_unsigned(_source: TransactionSource, call: &Self::Call) -> TransactionValidity {
 			let payload = Self::filter_valid_calls(&call)?;
 			Self::validate_signatures(&payload)?;
-
 			let nonce_check = PasskeyNonce::new(payload.passkey_call.clone());
-			nonce_check.validate()
+			nonce_check.validate()?;
+			let tx_charge = ChargeTransactionPayment::<T>(
+				payload.passkey_call.account_id.clone(),
+				call.clone(),
+			);
+			tx_charge.validate()
```

Comment:
I think we might want to return the nonce validate result since that has the dependency graph in it. But it would nice to take a look how the signed extension does this

Reply:
Figured from substrate that we need to combine the validity from all three checks via `ValidTransaction::combine_with`

here is how signed extensions are validated

https://github.com/paritytech/polkadot-sdk/blob/18ed309a37036db8429665f1e91fb24ab312e646/substrate/primitives/runtime/src/traits.rs#L1641